Parsers/ASimFileEvent/Parsers/vimFileEventM365D.yaml

```diff
@@ -15,115 +15,193 @@ References:
-       ActorUserIdType = iff(isempty(ActorUserSid),'AADID','SID'),
-       TargetFilePath = strcat(FolderPath, iff(DvcOs == "Linux", "/", "\\"), TargetFileName),
-       TargetFilePathType = iff(DvcOs == "Linux", "Unix", "Windows Local"),
-       SrcFilePath = strcat(PreviousFolderPath, iff(DvcOs == "Linux", "/", "\\"), SrcFileName),
-       SrcFilePathType = iff(DvcOs == "Linux", "Unix", "Windows Local"),
-       Hash=coalesce(TargetFileSHA256, TargetFileSHA1, TargetFileMD5),
-       NetworkApplicationProtocol = iff (RequestProtocol in (protocols), toupper(RequestProtocol), "")
-     | invoke _ASIM_ResolveDvcFQDN ('DeviceName')
-     | project-away DeviceName
-     | extend
-         HashType = tostring(dynamic(["SHA256", "SHA1", "MD5"])[array_index_of(pack_array(TargetFileSHA256, TargetFileSHA1, TargetFileMD5),Hash)])     
-     //  ****** Aliases
-     | extend 
-        User = ActorUsername,
-        Dvc = coalesce(DvcFQDN, DvcHostname),
-        FilePath = TargetFilePath,
-        Process = ActingProcessName,
-        CommandLine = ActingProcessCommandLine,
-        DvcMDEid = DvcId,
-        FileName = TargetFileName
-    | project-away MachineGroup, ReportId, SourceSystem, Initiating*, Timestamp, TenantId, Request*, PreviousFolderPath, FolderPath, AppGuardContainerId
-    | project-away ShareName, IsAzureInfoProtectionApplied, FileOrigin*, Sensitivity*
-    };
-    parser (disabled = disabled)
\ No newline at end of file
+  let parser=(
+      starttime: datetime=datetime(null),
+      endtime: datetime=datetime(null),
+      eventtype_in: dynamic=dynamic([]),
+      srcipaddr_has_any_prefix: dynamic=dynamic([]),
+      actorusername_has_any: dynamic=dynamic([]),
+      targetfilepath_has_any: dynamic=dynamic([]),
+      srcfilepath_has_any: dynamic=dynamic([]),
+      hashes_has_any: dynamic=dynamic([]),
+      dvchostname_has_any: dynamic=dynamic([]),
+      disabled: bool=false
+      ) {
+      let remote_events = 
+          DeviceFileEvents
+          | where not(disabled)
+          | where (isnull(starttime) or TimeGenerated >= starttime) 
+              and (isnull(endtime) or TimeGenerated <= endtime)
+          | where isnotempty(RequestAccountName)
+          | where ((array_length(eventtype_in) == 0 or ActionType in~ (eventtype_in))) and
+              ((array_length(srcipaddr_has_any_prefix) == 0) or (has_any_ipv4_prefix(RequestSourceIP, srcipaddr_has_any_prefix))) and 
+              ((array_length(actorusername_has_any) == 0) or (RequestAccountName has_any (actorusername_has_any)) or (RequestAccountDomain has_any (actorusername_has_any)) or (strcat(RequestAccountDomain, '\\', RequestAccountName) has_any (actorusername_has_any))) and
+              ((array_length(targetfilepath_has_any) == 0) or (FileName has_any (targetfilepath_has_any)) or (FolderPath has_any (targetfilepath_has_any)) or (strcat(FolderPath, iff(FolderPath startswith "/", "/", "\\"), FileName) has_any (targetfilepath_has_any))) and 
```

Comment:
Some of these filter conditions are not matching with actual field normalization. For instance actorusername_has_any filter applied on RequestAccountName , RequestAccountDomain  and               ((array_length(actorusername_has_any) == 0) or (RequestAccountName has_any (actorusername_has_any)) or (RequestAccountDomain has_any (actorusername_has_any)) or (strcat(RequestAccountDomain, '\\', RequestAccountName) has_any (actorusername_has_any))) and 
But while normalizing the ActorUseName, such conditions are not considered. Any reason for mismatch ?

Reply:
@anki-narravula RequestAccountName, RequestAccountDomain fields are used for building ActorWindowsUsername in the 'remote_events' section which later is mapped with ActorUsername using coalesce

Reply:
post filtering is not needed in this case as we are checking all possible fields in the pre-filtering check.